pkg/cnab/config_adapter/adapter_test.go

```diff
@@ -340,3 +341,69 @@ func TestPorter_generateDependencies(t *testing.T) {
 		})
 	}
 }
+
+func TestManifestConverter_GenerateCustomActionDefinitions(t *testing.T) {
+	c := config.NewTestConfig(t)
+	c.TestContext.AddTestFile("testdata/porter-with-custom-action.yaml", config.Name)
+
+	err := c.LoadManifest()
+	require.NoError(t, err)
+
+	a := ManifestConverter{
+		Context:  c.Context,
+		Manifest: c.Manifest,
+	}
+
+	defs := a.generateCustomActionDefinitions()
+	require.Len(t, defs, 2, "expected 2 custom action definitions to be generated")
+
+	require.Contains(t, defs, "status")
+	statusDef := defs["status"]
+	assert.Equal(t, "Prints out status of world", statusDef.Description)
+	assert.True(t, statusDef.Stateless, "expected the status custom action to be stateless")
+	assert.False(t, statusDef.Modifies, "expected the status custom action to not modify resources")
+
+	require.Contains(t, defs, "zombies")
+	zombieDef := defs["zombies"]
+	assert.Equal(t, "zombies", zombieDef.Description)
+	assert.False(t, zombieDef.Stateless, "expected the zombies custom action to default to not stateless")
+	assert.True(t, zombieDef.Modifies, "expected the zombies custom action to default to modifying resources")
+}
+
+func TestManifestConverter_generateDefaultAction(t *testing.T) {
+	a := ManifestConverter{}
+
+	testcases := []struct {
+		action     string
+		wantAction bundle.Action
+	}{
+		{"dry-run", bundle.Action{
```

Comment:
Should we add a test case testing the default (meaning, the `default` case of the `switch` statement).  Perhaps with an action named... `zombies`?!